raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -957,29 +976,6 @@ private void maybeSendOrHandleAppendRequest(long currentTimeMs) {
         }
     }
 
-    /**
-     * Append a set of records to the log. Successful completion of the future indicates a success of
-     * the append, with the uncommitted base offset and epoch.
-     *
-     * @param records The records to write to the log
-     * @return The uncommitted base offset and epoch of the appended records
-     */
-    @Override
-    public CompletableFuture<OffsetAndEpoch> append(Records records) {
```

Comment:
Rather than removing this, could we make it private? Seems better than having a bunch of callback logic nested in the constructor.